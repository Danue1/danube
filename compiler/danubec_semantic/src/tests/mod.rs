use crate::{Context, semantic};

#[test]
fn all_files() {
    let context = Context {
        path: std::path::PathBuf::from("src/tests/fixtures/lib.dnb"),
    };
    let (fs, table, symbols, diagnostic) = semantic(context);

    insta::assert_debug_snapshot!(fs);
    insta::assert_debug_snapshot!(table);
    insta::assert_debug_snapshot!(symbols);
    insta::assert_debug_snapshot!(diagnostic);
}
