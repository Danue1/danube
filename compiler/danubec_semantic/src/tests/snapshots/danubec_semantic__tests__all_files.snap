---
source: compiler/danubec_semantic/src/tests/mod.rs
expression: fs
---
Fs {
    inner: Inner {
        paths: SlotMap {
            slots: [
                Slot {
                    version: 0,
                    next_free: 0,
                },
                Slot {
                    version: 1,
                    value: "src/tests/fixtures/lib.dnb",
                },
                Slot {
                    version: 1,
                    value: "src/tests/fixtures/imports.dnb",
                },
            ],
            free_head: 3,
            num_elems: 2,
            _k: PhantomData<fn(danubec_symbol::FileId) -> danubec_symbol::FileId>,
        },
        canonicals: SecondaryMap {
            slots: [
                Vacant,
                Occupied {
                    value: "/Users/danuel/Documents/internal/danube-lang/danube/compiler/danubec_semantic/src/tests/fixtures/lib.dnb",
                    version: 1,
                },
                Occupied {
                    value: "/Users/danuel/Documents/internal/danube-lang/danube/compiler/danubec_semantic/src/tests/fixtures/imports.dnb",
                    version: 1,
                },
            ],
            num_elems: 2,
            _k: PhantomData<fn(danubec_symbol::FileId) -> danubec_symbol::FileId>,
        },
        files: SecondaryMap {
            slots: [
                Vacant,
                Occupied {
                    value: "mod imports;\n\nmod bar {\n  mod baz {\n    mod qux {\n      //\n    }\n  }\n}\n",
                    version: 1,
                },
                Occupied {
                    value: "use *;\nuse a;\nuse {};\nuse {a::{b::{c::d}}};\n",
                    version: 1,
                },
            ],
            num_elems: 2,
            _k: PhantomData<fn(danubec_symbol::FileId) -> danubec_symbol::FileId>,
        },
    },
}
